sormas-ui/src/main/java/de/symeda/sormas/ui/hospitalization/HospitalizationForm.java

```diff
@@ -225,6 +239,26 @@ protected void addFields() {
 				I18nProperties.getValidationError(Validations.beforeDate, intensiveCareUnitEnd.getCaption(), dischargeDateField.getCaption())));
 		hospitalizedPreviouslyField.addValueChangeListener(e -> updatePrevHospHint(hospitalizedPreviouslyField, previousHospitalizationsField));
 		previousHospitalizationsField.addValueChangeListener(e -> updatePrevHospHint(hospitalizedPreviouslyField, previousHospitalizationsField));
+		reasonForHospitalization.addValueChangeListener(this::updateOtherReasonForHospitalizationField);
+	}
+
+	private void updateOtherReasonForHospitalizationField(final Property.ValueChangeEvent event) {
```

Comment:
We have helper methods to replace custom logic like this. Please have a look at `FieldHelper.setVisibleWhen`, you can use this to make sure that "otherHospitalizationReason" is only visible when HospitalizationType.OTHER is selected, and that it gets cleared if another HospitalizationType is selected